Microsoft.Toolkit.Uwp.UI.Controls/HamburgerMenu/HamburgerMenu.cs

```diff
@@ -10,22 +10,42 @@
 // THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
 // ******************************************************************
 
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using Windows.Foundation.Metadata;
 using Windows.UI.Xaml;
 using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Data;
 
 namespace Microsoft.Toolkit.Uwp.UI.Controls
 {
     /// <summary>
     /// The HamburgerMenu is based on a SplitView control. By default it contains a HamburgerButton and a ListView to display menu items.
     /// </summary>
     [TemplatePart(Name = "HamburgerButton", Type = typeof(Button))]
-    [TemplatePart(Name = "ButtonsListView", Type = typeof(ListViewBase))]
-    [TemplatePart(Name = "OptionsListView", Type = typeof(ListViewBase))]
+    [TemplatePart(Name = "ButtonsListView", Type = typeof(Windows.UI.Xaml.Controls.ListViewBase))]
+    [TemplatePart(Name = "OptionsListView", Type = typeof(Windows.UI.Xaml.Controls.ListViewBase))]
+    [Obsolete("The HamburgerMenu will be removed in a future major release. Please use the NavigationView control available in the Fall Creators Update")]
     public partial class HamburgerMenu : ContentControl
     {
+        private static bool? _isNavViewSupported;
+
         private Button _hamburgerButton;
-        private ListViewBase _buttonsListView;
-        private ListViewBase _optionsListView;
+        private Windows.UI.Xaml.Controls.ListViewBase _buttonsListView;
+        private Windows.UI.Xaml.Controls.ListViewBase _optionsListView;
+
+        private ControlTemplate _previousTemplateUsed;
+        private object _navigationView;
+
+        private bool UsingNavView => UseNavigationViewWhenPossible && IsNavigationViewSupported;
+
+        /// <summary>
+        /// Gets a value indicating whether <see cref="NavigationView"/> is supported
+        /// </summary>
+        public static bool IsNavigationViewSupported => (bool)(_isNavViewSupported ??
```

Comment:
This should be 
`public static bool IsNavigationViewSupported { get; } = (bool)(_isNavViewSupported ??
            (_isNavViewSupported = ApiInformation.IsApiContractPresent("Windows.Foundation.UniversalApiContract", 5)));`
So it is called only once and the result is cached not every time the property get called

Reply:
Removed _isNavViewSupported completely